### ISSUE TITLE
jsx-a11y/label-has-for false negative for component named `Label`

### ISSUE THREAD

**Issue by emplums** [evcohen/eslint-plugin-jsx-a11y#638]
Hey 👋 

I'm using `jsx-a11y` in my component library, and I noticed that I get a false negative for a component called `Label` which is actually not a form label and renders as a `span`. `Label` is just a name for a UI pattern we've got in our design system that doesn't have anything to do with forms.

My code:
```
  <LabelGroup>
    <Label>Default label</Label>
    <Label>Darker gray label</Label>
    <Label outline>Default outline label</Label>
  </LabelGroup>
```

The `Label` render function:
```
const Label = styled('span')`
  display: inline-block;
  font-weight: 600;
  line-height: ${theme.lineHeights.condensedUltra};
  color: ${colors.white};
  border-radius: 2px;
  &:hover {
    text-decoration: none;
  }
```

as you can see it's just a `span` with some CSS.

When trying to lint my tests, I get eslint errors for using this component without a `htmlFor` attribute. I've tried using `eslint-disable-next-line jsx-a11y/label-has-for` to just silence these linting errors for now but it doesn't seem to be respected

**Comment by ljharb** [comment]
What's your eslint config? Is "Label" set up as a pseudo-label component name?

**Comment by emplums** [comment]
It is not! I just realized I need to do this, thanks!
